### PR TITLE
Fix whitespace buildup

### PR DESCRIPTION
**Pull request by jdkandersson**
Applicable spec: N/A

### Overview

Strips whitespace around user input content to avoid whitespace accumulation

### Rationale

Currently there is an issue where whitespace builds up continuously as it is not stripped from generated content before being added again.

### Module Changes

N/A

### Checklist

- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The documentation is generated using `src-docs`
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)
- [x] Changelog has been updated
- [ ] Version has been incremented

Version will be incremented on release

### CODE REVIEW

**Comment by deusebio** [comment]
The E2E tests seem to be happy about this change:

https://github.com/deusebio/gatekeeper-repo-test/actions/runs/7016270139/job/19087131474

File: src/reconcile.py [hunk 1]
```diff
@@ -678,8 +678,9 @@ def index_page(
         The action to take for the index page.
     """
     table_contents = "\n".join(table_row.to_markdown(discourse.host) for table_row in table_rows)
+    # Strip any whitespace around file contents to avoid building up more and more whitespace
```
Comment (varshigupta12) [inline src/reconcile.py:681]: Remove "more and more" just 'avoid building up whitespace' should be good

**Review by yanksyoon — APPROVED** [review]
LGTM!

File: src/reconcile.py (hunk 1 above)
Comment (jdkandersson) [reply src/reconcile.py]: Thanks, fixed

**Comment by jdkandersson** [comment]
Ok, everything passes now 💪

**Review by weiiwang01 — APPROVED** [review]

**Review by arturo-seijas — APPROVED** [review]

**Review by yanksyoon — APPROVED** [review]

**Pull request merged**